### DISCUSSION TITLE
In what editors is the Citations & Footnotes autocomplete feature available?

### DISCUSSION THREAD

**Discussion by pakiessling** [quarto-dev/quarto-cli#4048] [Q&A]
Hi, I am considering using quarto to write scientific reports.
I am wondering if the citation autocomplete feature, (getting suggestions after typing @), is exclusive to R-studio or could also work in other editors via a plugin or somethign?

Thank you!

**Comment by dragonstyle** [comment]
The vscode extension for Quarto will also autocomplete using your bibliography, but it does support the very rich citation insert features that RStudio does (e.g. Zotero, DOI, remove sources like PubMed).

**Reply by pakiessling** [reply]
Thank you for your reply. RStudio seems quite nice. The only reason I am hesitant to use it, is that I work with python.

Do .qmd's in Rstudio still rely on Reticulate for python cells under the hood?

**Reply by dragonstyle** [reply]
That makes sense. In RStudio, if you use the UI affordances to run individual cells, that will use reticulate. If you simply render the document, it will not (since in the individual cell case, it is really RStudio handling the execution of the code whereas in rendering the document you're letting Quarto handle code execution which will respect the various engine settings).

**Reply by pakiessling** [reply]
That clears it up. Thank you!
